### PR TITLE
War Traits

### PR DESCRIPTION
**Pull request by nfaletra**
Fixed and/or added trait tiers, levels, and values.

Implemented Fencer
* Also added item mods for Fencer
Implemented Shield Def. Bonus
* Also added item mods for Shield Def. Bonus

Gutted some sneaky *_MAGIC_SKILL globals that used to be in magic.lua and were still being referenced in a multitude of spell scripts.

### CODE REVIEW

**Comment by nfaletra** [comment]
Rebased after the damn bot changed some names.

**Comment by Cryptexx** [comment]
ELEMENTAL_MAGIC_SKILL -> dsp.skill.ELE

And what was wrong with it before?

**Comment by nfaletra** [comment]
ELEMENTAL_MAGIC_SKILL doesn't exist anymore. Skill values were tabled. Zynjec updated the skill names to be more readable. Updating the values to the new names now.

File: scripts/globals/spells/aspir.lua [hunk 1]
```diff
@@ -14,12 +14,12 @@ end;
...
     --calculate raw damage (unknown function  -> only dark skill though) - using http://www.bluegartr.com/threads/44518-Drain-Calculations
     -- also have small constant to account for 0 dark skill
-    local dmg = 5 + 0.375 * caster:getSkillLevel(DARK_MAGIC_SKILL);
+    local dmg = 5 + 0.375 * caster:getSkillLevel(dsp.skill.DRK);
     --get resist multiplier (1x if no resist)
     local params = {};
     params.diff = caster:getStat(dsp.mod.INT)-target:getStat(dsp.mod.INT);
     params.attribute = dsp.mod.INT;
-    params.skillType = DARK_MAGIC_SKILL;
+    params.skillType = dsp.skill.DRK;
```
Comment (TeoTwawki) [inline scripts/globals/spells/aspir.lua]: `dsp.skill.DARK_MAGIC`

File: scripts/globals/spells/aquaveil.lua [hunk 2]
```diff
@@ -21,7 +21,7 @@ function onSpellCast(caster,target,spell)
 
     local duration = 300;
     local power = AQUAVEIL_COUNTER + caster:getMod(dsp.mod.AQUAVEIL_COUNT);
-    if (caster:getSkillLevel(ENHANCING_MAGIC_SKILL) >= 200) then -- cutoff point is estimated. https://www.bg-wiki.com/bg/Aquaveil
+    if (caster:getSkillLevel(dsp.skill.ENH) >= 200) then -- cutoff point is estimated. https://www.bg-wiki.com/bg/Aquaveil
```
Comment (TeoTwawki) [inline scripts/globals/spells/aquaveil.lua]: `dsp.skill.ENHANCING_MAGIC`

File: scripts/globals/weaponskills.lua [hunk 3]
```diff
@@ -759,6 +769,16 @@ end;
         end
 
         nativecrit = nativecrit + (attacker:getMod(dsp.mod.CRITHITRATE)/100) + attacker:getMerit(dsp.merit.CRIT_HIT_RATE)/100 - target:getMerit(dsp.merit.ENEMY_CRIT_RATE)/100;
+        
+        -- Handle Fencer
+        local mainEquip = attacker:getStorageItem(0, 0, dsp.slot.MAIN);
+        local subEquip = attacker:getStorageItem(0, 0, dsp.slot.SUB);
+        if (mainEquip:isTwoHanded() == false and mainEquip:isHandToHand() == false) then
+            if (subEquip:getSkillType() == dsp.skill.NON or subEquip:isShield()) then
```
Comment (TeoTwawki) [inline scripts/globals/weaponskills.lua]: its `dsp.skill.NONE` now

**Pull request merged**